### PR TITLE
PR: Experiment with Angular signals #32 

### PR DESCRIPTION
**Pull request by tchojnacki**

### CODE REVIEW

File: code/angular.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
       "prefix": "app",
```
Comment (tchojnacki) [inline code/angular.json:11]: Changed automatically by Angular update migration tool.

File: code/package-lock.json
Comment (tchojnacki) [inline code/package-lock.json:1]: Changed automatically by Angular update migration tool.

File: code/package.json
Comment (tchojnacki) [inline code/package.json:1]: Changed automatically by Angular update migration tool.

File: code/projects/app/src/app/app.component.ts
Comment (tchojnacki) [inline code/projects/app/src/app/app.component.ts:1]: Changed automatically by Angular update migration tool.

File: code/projects/app/src/app/shared/file-download/file-download.component.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
   selector: "app-file-download",
-  standalone: true,
-  imports: [AsyncPipe, DownloadPipe],
   templateUrl: "./file-download.component.html",
   styleUrl: "./file-download.component.css",
 })
 export class FileDownloadComponent {
-  @Input({ required: true }) public file$!: Observable<File>;
+  public readonly file = input<File>();
+  protected readonly download = computedOpt(this.file, (f) => ({
```
Comment (tchojnacki) [inline code/projects/app/src/app/shared/file-download/file-download.component.ts:11]: Inlined pipe.

File: code/projects/app/src/app/shared/image-display/image-display.component.ts [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
-  @ViewChild("canvas", { static: true })
-  protected readonly canvas!: ElementRef<HTMLCanvasElement>;
+  private readonly canvas =
+    viewChild.required<ElementRef<HTMLCanvasElement>>("canvas");
 
-  public ngOnInit(): void {
-    this.image$.subscribe((image) => drawOn(image, this.canvas.nativeElement));
+  public constructor() {
+    effect(() => {
+      drawOn(this.image(), this.canvas().nativeElement);
```
Comment (tchojnacki) [inline code/projects/app/src/app/shared/image-display/image-display.component.ts:17]: Additionally, passing `undefined` to clear the `canvas` is now supported.

File: code/projects/app/src/app/stages/e1/e1.component.ts
Comment (tchojnacki) [inline code/projects/app/src/app/stages/e1/e1.component.ts:1]: Changed automatically by Angular update migration tool.

File: code/projects/app/src/app/stages/e2/e2.component.ts
Comment (tchojnacki) [inline code/projects/app/src/app/stages/e2/e2.component.ts:1]: Changed automatically by Angular update migration tool.

File: code/projects/app/src/app/stages/e3/slider/slider.component.html
Comment (tchojnacki) [inline code/projects/app/src/app/stages/e3/slider/slider.component.html:1]: **TODO:** This component is now just a mindless proxy to `input [(ngModel)]` and should be removed in the future.

File: code/projects/app/src/app/stages/e4/e4.component.ts
Comment (tchojnacki) [inline code/projects/app/src/app/stages/e4/e4.component.ts:1]: Changed automatically by Angular update migration tool.

**Review by mlodybercik — APPROVED** [review]
Please don't make me regret this decision.

**Pull request merged**